diff --git a/node_modules/expo-audio/ios/MediaController.swift b/node_modules/expo-audio/ios/MediaController.swift
index 2dd3fb2..5bd9443 100644
--- a/node_modules/expo-audio/ios/MediaController.swift
+++ b/node_modules/expo-audio/ios/MediaController.swift
@@ -37,8 +37,7 @@ class MediaController {
     }
     var nowPlayingInfo = nowPlayingInfoCenter.nowPlayingInfo ?? [String: Any]()
 
-    nowPlayingInfo[MPMediaItemPropertyPlaybackDuration] = player.duration
-    nowPlayingInfo[MPNowPlayingInfoPropertyElapsedPlaybackTime] = player.currentTime
+    nowPlayingInfo[MPNowPlayingInfoPropertyIsLiveStream] = true
     nowPlayingInfo[MPNowPlayingInfoPropertyPlaybackRate] = player.isPlaying ? player.ref.rate : 1.0
     nowPlayingInfo[MPNowPlayingInfoPropertyMediaType] = MPNowPlayingInfoMediaType.audio.rawValue
 
